src/diffusers/models/unet_2d_condition.py

```diff
@@ -589,6 +591,17 @@ def forward(
 
             down_block_res_samples += res_samples
 
+        if down_block_additional_residuals is not None:
+            new_down_block_res_samples = ()
+
+            for down_block_res_sample, down_block_additional_residual in zip(
+                down_block_res_samples, down_block_additional_residuals
+            ):
+                down_block_res_sample += down_block_additional_residual
+                new_down_block_res_samples += (down_block_res_sample,)
+
+            down_block_res_samples = new_down_block_res_samples
```

Comment:
Maybe we could do the same without creating a copy? (Not sure if it's important)

Reply:
Think it's fine - the memory overhead is insignificant here IMO